src/elf/dyn.rs

```diff
@@ -317,35 +317,35 @@ if_alloc! {
     }
 
     impl<'a> ctx::TryFromCtx<'a, Ctx> for Dyn {
-        type Error = ::error::Error;
+        type Error = crate::error::Error;
         type Size = usize;
         fn try_from_ctx(bytes: &'a [u8], Ctx { container, le}: Ctx) -> result::Result<(Self, Self::Size), Self::Error> {
             use scroll::Pread;
-            let dyn = match container {
+            let r#dyn = match container {
                 Container::Little => {
                     (bytes.pread_with::<dyn32::Dyn>(0, le)?.into(), dyn32::SIZEOF_DYN)
                 },
                 Container::Big => {
                     (bytes.pread_with::<dyn64::Dyn>(0, le)?.into(), dyn64::SIZEOF_DYN)
                 }
             };
-            Ok(dyn)
+            Ok(r#dyn)
         }
     }
 
     impl ctx::TryIntoCtx<Ctx> for Dyn {
-        type Error = ::error::Error;
+        type Error = crate::error::Error;
         type Size = usize;
         fn try_into_ctx(self, bytes: &mut [u8], Ctx { container, le}: Ctx) -> result::Result<Self::Size, Self::Error> {
             use scroll::Pwrite;
             match container {
                 Container::Little => {
-                    let dyn: dyn32::Dyn = self.into();
-                    Ok(bytes.pwrite_with(dyn, 0, le)?)
+                    let r#dyn: dyn32::Dyn = self.into();
+                    Ok(bytes.pwrite_with(r#dyn, 0, le)?)
                 },
```

Comment:
This file and these variables need to be renamed. Could you suggest a new name for it?

Reply:
Wow the file needs to be renamed ? Why? This will be a backwards incompatible change :(

Reply:
`dyn` is keyword in 2018 edition. You still could use it by raw keyword `r#dyn`.

Reply:
Oh yea. And I guess it makes sense we can’t use keywords in filenames because they become nodules ? :(

@philipc what do you think ? Probably rename to Dynamic or something :(

I’m sad

Reply:
Yes, we can. But when we import module, we must write something like `use parent::r#dyn::Object` instead.

Reply:
We can keep deprecated aliases for the old names if you want to keep backwards compatibility. I do think we need to rename it though, using `r#dyn` in 2018 edition code isn't nice, that syntax is intended for interop with 2015 edition code. I don't have any better suggestions than `Dynamic`.

Reply:
I would do the rename in separate PR and update CHANGELOG in there.

Reply:
Is Dyn also banned (capital C)?

So I suggest renaming to dyn_syms or dynamic (I don’t like dynamic personally), but perhaps we should bikeshed later

Reply:
`Dyn` doesn't need to change.

The `dyn` module is for the `PT_DYNAMIC` segment/`.dynamic` section, not for dynamic symbols.

Reply:
Discussion will be moved to #135.